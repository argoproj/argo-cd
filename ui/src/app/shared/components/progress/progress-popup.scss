.progress-popup__title {
    font-family: "Heebo", sans-serif;
    font-weight: normal;
    font-size: 16px;
    font-style: normal;
    color: #495763;
}

.progress-popup__footer {
    text-align: right;
}

.progress-popup__content {
    padding-top: 15px;
}
