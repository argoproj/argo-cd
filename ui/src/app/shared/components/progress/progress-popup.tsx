import {Popup} from 'argo-ui';
import * as React from 'react';
import {ProgressBar} from './progress-bar';

const Title = ({title}: {title: string}) => {
    return <React.Fragment>{title}</React.Fragment>;
};

const Footer = ({percentage, onClose}: {percentage: number; onClose: () => void}) => {
    return (
        <div style={{textAlign: 'right'}}>
            {percentage >= 100 && (
                <button className='argo-button argo-button--base-o' onClick={() => onClose()}>
                    Close
                </button>
            )}
        </div>
    );
};

export const ProgressPopup = ({title, percentage, onClose}: {title: string; percentage: number; onClose: () => void}) => {
    return (
        <Popup title={<Title title={title} />} footer={<Footer percentage={percentage} onClose={onClose} />}>
            <ProgressBar percentage={percentage} />
        </Popup>
    );
};
