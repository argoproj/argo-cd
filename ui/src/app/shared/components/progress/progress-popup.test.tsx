import * as renderer from 'react-test-renderer';
import * as React from 'react';
import {ProgressPopup} from './progress-popup';

test('ProgressPopup.0%', () => {
    const state = renderer.create(<ProgressPopup onClose={() => {
    }} percentage={0} title={''}/>);

    expect(state).toMatchSnapshot();
});

test('ProgressPopup.50%', () => {
    const state = renderer.create(<ProgressPopup onClose={() => {
    }} percentage={50} title={'My Title'}/>);

    expect(state).toMatchSnapshot();
});

test('ProgressPopup.100%', () => {
    const state = renderer.create(<ProgressPopup onClose={() => {
    }} percentage={100} title={''}/>);

    expect(state).toMatchSnapshot();
});
