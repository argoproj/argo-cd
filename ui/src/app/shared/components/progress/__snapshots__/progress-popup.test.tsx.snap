// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressPopup.0% 1`] = `
<div
  className="popup-overlay"
>
  <div
    className="popup-container"
  >
    <div
      className="row popup-container__header"
    >
      
    </div>
    <div
      className="row popup-container__body"
    >
      <div
        className="columns large-12"
      >
        <div
          style={
            Object {
              "background": "#CCD6DD",
              "borderRadius": "5px",
              "height": "10px",
              "margin": "15px auto",
              "width": "90%",
            }
          }
        >
          <div
            style={
              Object {
                "background": "#18BE94",
                "borderRadius": "inherit",
                "height": "100%",
                "transition": "width 0.1s ease-in",
                "width": "0%",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="row popup-container__footer"
    >
      <div
        style={
          Object {
            "textAlign": "right",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`ProgressPopup.50% 1`] = `
<div
  className="popup-overlay"
>
  <div
    className="popup-container"
  >
    <div
      className="row popup-container__header"
    >
      My Title
    </div>
    <div
      className="row popup-container__body"
    >
      <div
        className="columns large-12"
      >
        <div
          style={
            Object {
              "background": "#CCD6DD",
              "borderRadius": "5px",
              "height": "10px",
              "margin": "15px auto",
              "width": "90%",
            }
          }
        >
          <div
            style={
              Object {
                "background": "#18BE94",
                "borderRadius": "inherit",
                "height": "100%",
                "transition": "width 0.1s ease-in",
                "width": "100%",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="row popup-container__footer"
    >
      <div
        style={
          Object {
            "textAlign": "right",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`ProgressPopup.100% 1`] = `
<div
  className="popup-overlay"
>
  <div
    className="popup-container"
  >
    <div
      className="row popup-container__header"
    >
      
    </div>
    <div
      className="row popup-container__body"
    >
      <div
        className="columns large-12"
      >
        <div
          style={
            Object {
              "background": "#CCD6DD",
              "borderRadius": "5px",
              "height": "10px",
              "margin": "15px auto",
              "width": "90%",
            }
          }
        >
          <div
            style={
              Object {
                "background": "#18BE94",
                "borderRadius": "inherit",
                "height": "100%",
                "transition": "width 0.1s ease-in",
                "width": "100%",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="row popup-container__footer"
    >
      <div
        style={
          Object {
            "textAlign": "right",
          }
        }
      >
        <button
          className="argo-button argo-button--base-o"
          onClick={[Function]}
        >
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;
